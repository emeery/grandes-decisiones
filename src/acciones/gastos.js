import uuid from 'uuid';

export const crearGasto = ({
    descripcion = '', 
    nota = '',
    monto = 0,
    date = 0
 }) => ({
    type: 'SUMA_GASTO',
    gasto: {
        id: uuid(),
        descripcion,
        nota,
        monto,
        date
    }
});
// 
export const removerGasto = ({ id } = {}) => ({
    type: 'REMUEVE_GASTO',
    id
});
// editar descripcion o monto, 
export const editarGasto = (id, update) => ({
    type: 'EDITA_GASTO',
    id,
    update
});