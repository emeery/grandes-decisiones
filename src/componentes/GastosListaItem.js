import React from 'react';
import { Link } from 'react-router-dom';
//import {connect } from 'react-redux';
// import {removerGasto} from '../acciones/gastos'; connect 

// destructuracion dispatch
        
//dispatch, id, descripcion, monto, fecha
const GastosListaItem = ({
    id, descripcion, monto, date}) => (
    <div>
        <Link to={`/edita/${id}`}>
        <h3> {descripcion} </h3>
        </Link>
        <h2> {monto} - {date}</h2>
       
    </div>
);
export default GastosListaItem;

// const GastosListaItem = ({ 
//      dispatch, id, descripcion, monto, fecha}) => (
//      <div>
//         <h3> descripcion </h3>
//         <h2>fecha: {fecha} - monto: {monto}</h2>
//         <button onClick={() => {
//             dispatch(removeGasto({ id }));
//         }}
//         > remover </button>
//     </div>
// );
// export default connect()(GastosListaItem);

// <h2> monto fecha </h2>
// <button onClick={() => {
//     dispatch(removerGasto({ id }));
// }}
// > remover </button>