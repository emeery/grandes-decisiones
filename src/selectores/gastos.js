import moment from 'moment';
// valida las fechas y el texto /
// seleccionaGasto
export default (gastos, {texto, ordenarPor, startDate, endDate}) => {
    return gastos.filter((gasto) => {
        const creadoEnElMomento = moment(gasto.date);
        const fechaInicioMatch = startDate ? startDate.isSameOrBefore(creadoEnElMomento, 'day') : true;
        const fechaFinMatch = endDate ? endDate.isSameOrAfter(creadoEnElMomento, 'day'): true;
        const textoMatch = gasto.descripcion.toLowerCase().includes(texto);
        return textoMatch && fechaInicioMatch && fechaFinMatch;
    }).sort(( a , b )=> {
        if (ordenarPor === 'date') {
            return a.date < b.date ? 1 : -1;
        } else if (ordenarPor === 'monto') {
            return a.monto < b.monto ? 1 : -1;
        }
    });
};

// The selectors are functions that take the status of Redux 
// as an argument and return some data to pass to the component.

// const fechaInicioMatch = typeof fechaInicio !== 'number' || gasto.fecha >= fechaInicio;
// const fechaFinMatch = typeof fechaFin !== 'number' || gasto.fecha <= fechaFinMatch;
