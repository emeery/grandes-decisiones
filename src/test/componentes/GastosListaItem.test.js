import React from 'react';
import {shallow} from 'enzyme';
import gastos from '../accesorios/gastos';
import GastosListaItem 
from '../../componentes/GastosListaItem'
test(' el componente GastosListaItem',()=>{
    const envoltura = shallow(<GastosListaItem 
        {...gastos[1]}
        />
    );
    expect(envoltura).toMatchSnapshot();
});