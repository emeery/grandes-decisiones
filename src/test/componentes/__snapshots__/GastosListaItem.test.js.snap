// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` el componente GastosListaItem 1`] = `
<div>
  <Link
    replace={false}
    to="/edita/2"
  >
    <h3>
       
      renta
       
    </h3>
  </Link>
  <p>
    $1,095.00
    -
    December 27th, 1969
  </p>
</div>
`;
